crates/autopilot/src/domain/settlement/solution/trade.rs

```diff
@@ -271,8 +279,8 @@ impl Trade {
             fee::Policy::PriceImprovement {
                 factor,
                 max_volume_factor,
-                quote,
             } => {
+                let quote = quote.ok_or(Error::MissingQuote)?;
```

Comment:
Yes, previously, we could check that when building a protocol fee instance. But now, the protocol fee doesn't require a quote anymore since it is part of the order. There is still a chance to have an order without a quote, but that shouldn't happen for an order that contains protocol fees(PriceImprovement).